Fix telemetry modal

cwillisf opened the pull request:
### Resolves

Resolves #99 (again)
Resolves #102

### Proposed Changes

This change splits `ScratchDesktopHOC` / `ScratchDesktopComponent` into `ScratchDesktopOuterHOC` / `ScratchDesktopOuterComponent` and `ScratchDesktopInnerHOC` / `ScratchDesktopInnerComponent`.

`ScratchDesktopOuterComponent` is now responsible for `showTelemetryModal` which only works if it comes from outside the `AppStateHOC` since it's used in the `AppStateHOC` constructor. The outer component also handles a few static props, like `isScratchDesktop`.

`ScratchDesktopInnerComponent` handles everything else, most notably anything that interacts with the state established by `AppStateHOC`.

I've also added telemetry reset instructions to `README.md` as suggested in #102.

### Reason for Changes

The order of HOC composition -- specifically, `AppStateHOC` vs. `ScratchDesktopHOC` -- has gone back and forth a few times in the process of fixing issues. It turns out this is because neither order was correct: some properties that `ScratchDesktopHOC` supplies are needed by `AppStateHOC`, but some properties of `ScratchDesktopHOC` require state information that's provided by `AppStateHOC`.

This change splits those two categories into two separate HOCs, resolving that conflict.

### Test Coverage

- Tested that this doesn't break #150: Starting the app with a project file name on the command line still opens the project
- Tested that this fixes #99 by deleting `telemetry.json` and confirming that the telemetry modal does appear

cwillisf committed:
split HOC in two: one inside AppStateHOC, one out

`ScratchDesktopOuterComponent` is now responsible for
`showTelemetryModal` which only works if it comes from outside the
`AppStateHOC` since it's used in the `AppStateHOC` constructor. The
outer component also handles a few static props, like
`isScratchDesktop`.

`ScratchDesktopInnerComponent` handles everything else, most notably
anything that interacts with the state established by `AppStateHOC`.

cwillisf committed:
add telemetry reset instructions to README.md

cwillisf commented:
I ran into this while working on https://github.com/cwillisf/scratch-desktop/tree/privacy-policy which is almost ready for review...

rschamp commented on src/renderer/app.jsx:
```diff
@@ -43,8 +43,26 @@ document.body.appendChild(appTarget);
+        const shouldShowTelemetryModal = (typeof ipcRenderer.sendSync('getTelemetryDidOptIn') !== 'boolean');
+
+        return (<WrappedComponent
+            canEditTitle
+            canModifyCloudData={false}
+            isScratchDesktop
```
These 3 props seem like they belong in the inner HOC since one could see the role of this outer HOC as to pass props into `AppStateHOC`. If that's the case then I think it would be clearer if we only had things necessary for AppStateHOC in this outer HOC, maybe accompanied by a more specific name like `ScratchDesktopAppStateHOC`?

cwillisf committed:
clarify HOC names and move props to match

Previously the two HOCs in app.jsx were named according to their place
in the component structure, which didn't provide any information about
their functionality or meaning. Now they are named according to the
components they wrap, which should help with future maintenance in that
it will guide which props belong in each.

cwillisf committed:
Merge branch 'develop' into fix-telemetry-modal

cwillisf committed:
fix merge & rearrange components to satisfy lint
